src/keyfile.c

```diff
@@ -351,17 +351,40 @@ static void save_recent_files(GKeyFile *config, GQueue *queue, gchar const *key)
 }
 
 
-static gchar *get_session_file_string(GeanyDocument *doc)
+static gchar *get_session_file_string(GeanyDocument *doc, gchar * project_root_folder)
 {
 	gchar *fname;
+	gchar *relative_filename;
+	gchar *doc_filename;
 	gchar *locale_filename;
 	gchar *escaped_filename;
 	GeanyFiletype *ft = doc->file_type;
+	GFile *file_doc;
+	GFile *file_root_folder;
 
 	if (ft == NULL) /* can happen when saving a new file when quitting */
 		ft = filetypes[GEANY_FILETYPES_NONE];
 
-	locale_filename = utils_get_locale_from_utf8(doc->file_name);
+	/* project_root_folder contain a path only if the project use relative path */
+	if(project_root_folder){
+		file_root_folder = g_file_new_for_path(project_root_folder);
+		file_doc = g_file_new_for_path(doc->file_name);
+		relative_filename = g_file_get_relative_path(file_root_folder, file_doc);
+		if(relative_filename){
+			/* Append './' so we know it is a relative filename */
+			doc_filename = g_strconcat("./",relative_filename, NULL);
```

Comment:
As `relative_filename` is in locale encoding technically you don't know if `./` is properly encoded to prepend (not append)

Reply:
Make sense, I will change the behavior, I don't need to store "./" at the  beginning of the path.